samples/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -128,42 +116,108 @@ static int ble_adv_stop(void)
 	return err;
 }
 
-static int ble_adv_start(void)
+static void bond_scan(const struct bt_bond_info *info, void *user_data)
 {
-	const struct bt_le_adv_param *adv_param = &adv_param_normal;
-	int err;
+	struct bond_scan_data *bond_scan_data = user_data;
 
-	if (IS_ENABLED(CONFIG_DESKTOP_BLE_FAST_ADV)) {
-		err = bt_le_adv_stop();
-		if (err) {
-			LOG_ERR("Cannot stop advertising (err %d)", err);
-			return err;
-		}
+	if (bond_scan_data->peer_id == bond_scan_data->peer_count) {
+		bt_addr_le_copy(&bond_scan_data->peer_address, &info->addr);
+	}
+
+	__ASSERT_NO_MSG(bond_scan_data->peer_count < UCHAR_MAX);
+	bond_scan_data->peer_count++;
+}
+
+static int ble_adv_start_direct(const bt_addr_le_t *addr, bool fast_adv)
+{
+	const struct bt_le_adv_param *adv_param = BT_LE_ADV_CONN_DIR_LOW_DUTY;
 
+	if (fast_adv) {
 		LOG_INF("Use fast advertising");
+		adv_param = BT_LE_ADV_CONN_DIR;
+	}
+
+	struct bt_conn *conn = bt_conn_create_slave_le(addr, adv_param);
+
+	if (conn == NULL) {
+		return -EFAULT;
+	}
+
+	bt_conn_unref(conn);
```

Comment:
I checked and we unref this connection reference because we don't need it here, but `bt_conn_create_slave_le()` gives us a reference caller should manage. So the usage is ok here.

Reply:
thx